src/core/ctf.h

```diff
@@ -112,6 +120,18 @@ class CTF {
     float               ParticleShiftGivenAzimuth(float azimuth);
     float               WavelengthGivenAccelerationVoltage(float acceleration_voltage);
 
+    inline float xi(float squared_spatial_frequency) {
```

Comment:
This is the "Fresnel Propagator" - suggest naming it that or phase_shift_from_focal_difference or something. (I'm sure you saw it is the defocus term in the normal aberration eq., same thing used to propagate a small phase shift for a "slice" in the simulator)

Reply:
Not sure if I understand. I though the Fresnel propagator is the aberration caused by a very thin slice. Xi is the argument of the sinc function that comes out of integrating over a bunch of fresnel-propagators. (Why would the fresnel propagator care about the thickness of the sample?)

Maybe the best thing is to implement sinc somewhere else as you suggested and called sinc_xi the "thon_ring_modulator"

Reply:
Integrating the Fresnel propagator over the distance from the sample entry point, to the BFP is what gives the total defocus term. It is the integral that "cares" about the total thickness